File: Java/ex02.java
import java.util.Scanner;

public class ex02 {
    public static void main(String[] args){
        Scanner in = new Scanner(System.in);
        System.out.print("Digite um numero: ");
        long num = in.nextLong();

        if (num % 2 == 0){
            System.out.println("Eh par");
        }
        else{
            System.out.println("Eh impar");
        }

        in.close();
    }    
}

File: C/ex02.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    char input[20]; // pode aumentar pra ler mais números
    int num;
    char *endptr; 

    printf("Digite o numero: ");
    fgets(input, sizeof(input), stdin); 

    num = strtol(input, &endptr, 10);
    
    if (input[0] != '\n' && *endptr == '\n') {
        if (num % 2 != 0) {
            printf("%d se trata de um numero impar.\n", num);
        } else {
            printf("%d se trata de um numero par.\n", num);
        }
    } else {
        printf("Entrada invalida. Por favor, apenas numeros.\n");
    }

    return 0;
}
